libs/sdk-core/src/models.rs

```diff
@@ -550,6 +550,15 @@ pub enum PaymentTypeFilter {
     ClosedChannel,
 }
 
+/// A metadata filter which can be applied when retrieving the transaction list
+pub struct PaymentMetadataFilter {
+    /// Specifies which field to apply the filter on, using the JSON path format
+    pub search_path: String,
+    /// Specifies which JSON value to filter for.
+    /// As such, strings must be wrappe with quotes ("") in order to be properly filtered
```

Comment:
I don't understand the comment here. Should the user add quotes as part of the value here?

Reply:
Yes, if the value inside the JSON object is a JSON string, the developer must add double quotes in order to match for it. The `->` operator simply creates a one-to-one match, so "string" can only be matched by the literal "string".

Reply:
I see, I think it will create a confusion. Users will not be aware of that, they will just pass a string...
@yaslama any idea on how to make the experience here more straight forward?

Reply:
@hydra-yse @yaslama perhaps we can trim the quotes before the matching?

`trim(metadata->'$.{}', '"') = '{}'`

Reply:
@hydra-yse  Now thinking again my suggestion will make a problem in case the user pass json (nested).
Ok, I am fine with the quotes as we document the value should be a valid json (where strings should be quoted)